src/udp_wrap.cc

```diff
@@ -679,7 +683,16 @@ void UDPWrap::OnAlloc(uv_handle_t* handle,
 }
 
 uv_buf_t UDPWrap::OnAlloc(size_t suggested_size) {
-  return AllocatedBuffer::AllocateManaged(env(), suggested_size).release();
+  Environment* env = this->env();
+  uv_buf_t buf;
+  {
+    NoArrayBufferZeroFillScope no_zero_fill_scope(env->isolate_data());
+    std::unique_ptr<BackingStore> bs =
+        ArrayBuffer::NewBackingStore(env->isolate(), suggested_size);
+    buf = uv_buf_init(static_cast<char*>(bs->Data()), bs->ByteLength());
+    env->released_allocated_buffers()->emplace(buf.base, std::move(bs));
+  }
+  return buf;
```

Comment:
Please add utility functions for this (and for the expanded code in `::OnRecv` to the `Environment` class, otherwise we’ll duplicate this exact code into other functions like `EmitToJSStreamListener::OnStreamAlloc`.

Reply:
@addaleax moved the codes to the `Environment` class, PTAL